src/cli/cli_coap_secure.hpp

```diff
@@ -0,0 +1,127 @@
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file contains definitions for a simple CLI CoAP Secure (server and) client.
+ *   Note: Server functionality now not implemented.
+ */
+
+#ifndef CLI_COAP_SECURE_HPP_
+#define CLI_COAP_SECURE_HPP_
+
+#include "openthread-core-config.h"
+
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+
+#include <openthread/types.h>
+
+#include "coap/coap_header.hpp"
+#include "coap/coap_secure.hpp"
+
+namespace ot {
+namespace Cli {
+
+class Interpreter;
+
+/**
+ * This class implements the CLI CoAP Secure server and client.
+ *
+ */
+class CoapSecureCli
+{
+public:
+    /**
+     * Constructor
+     *
+     * @param[in]  aInterpreter  The CLI interpreter.
+     *
+     */
+    CoapSecureCli(Interpreter &aInterpreter);
+
+    /**
+     * This method interprets a list of CLI arguments.
+     *
+     * @param[in]  argc  The number of elements in argv.
+     * @param[in]  argv  A pointer to an array of command line arguments.
+     *
+     */
+    otError Process(int argc, char *argv[]);
+
+private:
+    enum
+    {
+        kMaxUriLength  = 32,
+        kMaxBufferSize = 16
+    };
+
+    void PrintHeaderInfos(otCoapHeader *aHeader) const;
+
+    void PrintPayload(otMessage *aMessage) const;
+
+    otError ProcessRequest(int argc, char *argv[]);
+
+    otError Stop(void);
+
+    static void OTCALL HandleServerResponse(void *               aContext,
+                                            otCoapHeader *       aHeader,
+                                            otMessage *          aMessage,
+                                            const otMessageInfo *aMessageInfo);
+    void HandleServerResponse(otCoapHeader *aHeader, otMessage *aMessage, const otMessageInfo *aMessageInfo);
+
+    static void OTCALL HandleClientResponse(void *               aContext,
+                                            otCoapHeader *       aHeader,
+                                            otMessage *          aMessage,
+                                            const otMessageInfo *aMessageInfo,
+                                            otError              aError);
+    void               HandleClientResponse(otCoapHeader *       aHeader,
+                                            otMessage *          aMessage,
+                                            const otMessageInfo *aMessageInfo,
+                                            otError              aError);
+
+    static void OTCALL HandleClientConnect(const bool aConnected, void *aContext);
+    void               HandleClientConnect(const bool aConnected);
+
+    otCoapResource mResource;
+    char           mUriPath[kMaxUriLength];
+    bool           mShutdownFlag;
+
+    Interpreter &mInterpreter;
+
+    uint8_t mPsk[32];
```

Comment:
nit: prefer using something like `OT_COAPS_PSK_MAX_LENGTH`

Reply:
can i add constant kPskMaxLength and kPskIdMaxLength to the enum instead?